File: Python/String-1/without2.py

# given a string, if a length 2 substring appears at both its beginning and end, return a string without the substring at the beginning
# so "HelloHe" yields "lloHe"
# the substring may overlap with itself, so "Hi" yields ""
# otherwise, return the original string unchanged
def without2(str):

    if len(str) == 2:
        s = ''
    elif len(str) > 2 and str[:2] == str[-2:]:
        s = str[2:]
    else:
        s = str

    return s

print(without2('HelloHe'))
print(without2('HelloHi'))
print(without2('Hi'))

File: JavaScript/String-1/without2.js

// given a string, if a length 2 substring appears at both its beginning and end, return a string without the substring at the beginning
// so 'HelloHe' yields 'lloHe'
// the substring may overlap with itself
// so 'Hi' yields '
// otherwise, return the original string unchanged
const without2 = (str) => {
	if (str.length > 1 && (str.substring(0, 2) === str.substring(str.length - 2, str.length)) {
		str = str.substring(2);
	}

	return str;
};

console.log(without2('HelloHe'));
console.log(without2('HelloHi'));
console.log(without2('Hi'));
